src/LinAlg/matrix.h

```diff
@@ -1345,11 +1348,12 @@ namespace utl //! General utility classes and functions.
   }
 
   template<class T> inline
-  vector<T>& vector<T>::add(const std::vector<T>& X, const T& alfa)
+  vector<T>& vector<T>::add(const std::vector<T>& X, const T& alfa,
+                            int ofsx, int stridex)
   {
     T* p = this->ptr();
-    const T* q = &X.front();
-    for (size_t i = 0; i < this->size() && i < X.size(); i++, p++, q++)
+    const T* q = &X.front() + ofsx;
```

Comment:
`x.data() + ofsx` ?

Maybe the `ptr()` method also should be replaced by `data()` which is inherited from `std::vector` and does the same. `ptr()` was added long ago and before `data()` was known to the author.

Reply:
sure.